File: src/main/java/com/github/kindrat/cassandra/client/ui/eventhandler/FilterBtnHandler.java
package com.github.kindrat.cassandra.client.ui.eventhandler;

import com.github.kindrat.cassandra.client.filter.DataFilter;
import com.github.kindrat.cassandra.client.ui.DataObject;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.scene.control.TableView;
import javafx.scene.control.TextField;
import lombok.RequiredArgsConstructor;

import java.util.List;
import java.util.function.Predicate;
import java.util.stream.Collectors;

@RequiredArgsConstructor
public class FilterBtnHandler implements EventHandler<ActionEvent> {
    private final TextField filterTb;
    private final TableView<DataObject> dataTable;
    private final ObservableList<DataObject> originalData;

    @Override
    public void handle(ActionEvent event) {
        if (filterTb.getText().isEmpty()) {
            if (dataTable.getItems().size() != originalData.size()) {
                dataTable.setItems(originalData);
            }
        } else {
            Predicate<DataObject> rowPredicate = DataFilter.parsePredicate(filterTb.getText());
            List<DataObject> filtered = originalData.stream().filter(rowPredicate).collect(Collectors.toList());
            dataTable.setItems(FXCollections.observableArrayList(filtered));
        }
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/eventhandler/FilterBtnHandler.kt
package com.github.kindrat.cassandra.client.ui.eventhandler

import com.github.kindrat.cassandra.client.filter.DataFilter.parsePredicate
import com.github.kindrat.cassandra.client.ui.DataObject
import javafx.collections.FXCollections
import javafx.collections.ObservableList
import javafx.event.ActionEvent
import javafx.event.EventHandler
import javafx.scene.control.TableView
import javafx.scene.control.TextField
import kotlin.streams.toList

class FilterBtnHandler(
        private val filterTb: TextField,
        private val dataTable: TableView<DataObject>,
        private val originalData: ObservableList<DataObject>
) : EventHandler<ActionEvent?> {

    override fun handle(event: ActionEvent?) {
        if (filterTb.text.isEmpty()) {
            if (dataTable.items.size != originalData.size) {
                dataTable.items = originalData
            }
        } else {
            val rowPredicate = parsePredicate(filterTb.text)
            val filtered = originalData.stream().filter(rowPredicate).toList()
            dataTable.setItems(FXCollections.observableArrayList(filtered))
        }
    }
}